starport/services/scaffolder/component.go

```diff
@@ -142,3 +150,83 @@ func isGoReservedWord(name string) bool {
 	}
 	return false
 }
+
+// ErrComponentAlreadyCreated is the error returned when a specific component is already created
+type ErrComponentAlreadyCreated struct {
+	name          string
+	checkedType   string
+	componentType string
+}
+
+// NewErrComponentAlreadyCreated returns a new ErrComponentAlreadyCreated error
+func NewErrComponentAlreadyCreated(name, checkedType, componentType string) *ErrComponentAlreadyCreated {
+	return &ErrComponentAlreadyCreated{
+		name,
+		checkedType,
+		componentType,
+	}
+}
+
+func (e *ErrComponentAlreadyCreated) Error() string {
+	return fmt.Sprintf("component %s with name %s is already created (type %s exists)",
+		e.componentType, e.name, e.checkedType)
+}
+
+// checkComponentCreated checks if the component has been already created with Starport in the project
+func checkComponentCreated(appPath, moduleName, compName string) (err error) {
+	compNameTitle := strings.Title(compName)
+
+	// associate the type to check with the component that scaffold this type
+	typesToCheck := map[string]string{
+		compNameTitle:                           componentType,
+		"MsgCreate" + compNameTitle:             componentType,
+		"MsgUpdate" + compNameTitle:             componentType,
+		"MsgDelete" + compNameTitle:             componentType,
+		"QueryAll" + compNameTitle + "Request":  componentType,
+		"QueryAll" + compNameTitle + "Response": componentType,
+		"QueryGet" + compNameTitle + "Request":  componentType,
+		"QueryGet" + compNameTitle + "Response": componentType,
+		"Msg" + compNameTitle:                   componentMessage,
+		"Query" + compNameTitle + "Request":     componentQuery,
+		"Query" + compNameTitle + "Response":    componentQuery,
+		"MsgSend" + compNameTitle:               componentPacket,
+		compNameTitle + "PacketData":            componentPacket,
+	}
+
+	absPath, err := filepath.Abs(filepath.Join(appPath, "x", moduleName, "types"))
+	if err != nil {
+		return err
+	}
+	fileSet := token.NewFileSet()
+	all, err := parser.ParseDir(fileSet, absPath, func(os.FileInfo) bool { return true }, parser.ParseComments)
+	if err != nil {
+		return err
+	}
+
+	for _, pkg := range all {
+		for _, f := range pkg.Files {
+			ast.Inspect(f, func(x ast.Node) bool {
+				typeSpec, ok := x.(*ast.TypeSpec)
+				if !ok {
+					return true
+				}
+
+				if _, ok := typeSpec.Type.(*ast.StructType); !ok {
+					return true
+				}
+
+				// Check if the parsed type is from a scaffolded component with the name
+				if compType, ok := typesToCheck[typeSpec.Name.Name]; ok {
+					err = NewErrComponentAlreadyCreated(compName, typeSpec.Name.Name, compType)
```

Comment:
Any reason why we need a custom error type here because err returned by `checkComponentCreated` is not asserted by the caller and `ErrComponentAlreadyCreated` has no public fields. Instead we could just use `fmt.Errorf`.

Reply:
I wanted to have some consistence with the previous return signature where a boolean was returned in case of already created component
In practice we directly the error returned by this function is not filtered but I think it's better to have this ability for maintainability

Reply:
If there is no a use case to differentiate error type by the caller, I think creating a custom error just adds more complexity. In other parts of `checkComponentValidity` we use `fmt.Errorf` and they're also about validating things. I doe.n't see why this one should have a special err type.